src/tests/translation/number_binop.rs

```diff
@@ -0,0 +1,701 @@
+use crate::{tests::translation::translate_amber_code, unwrap_fragment};
+use crate::modules::prelude::*;
+
+#[test]
+fn test_translation_add_int() {
+    let code = r"
+        echo 15 + 45
+        echo 39 + 21 + 80
+        echo 3 + 9 + 648 + 1232
+    ";
+
+    let ast = translate_amber_code(code).expect("Couldn't translate Amber code");
+
+    let block = unwrap_fragment!(ast, Block);
+    assert_eq!(block.statements.len(), 3);
+    {
+        let arith = unwrap_fragment!(unwrap_fragment!(block.statements[0].clone(), List).values[1].clone(), Arithmetic);
+        assert_eq!(arith.op, ArithOp::Add);
+        let left = unwrap_fragment!(arith.left.unwrap().clone(), Raw);
+        assert_eq!(left.value, "15");
+        let right = unwrap_fragment!(arith.right.unwrap().clone(), Raw);
+        assert_eq!(right.value, "45");
+    }
+    {
+        let arith = unwrap_fragment!(unwrap_fragment!(block.statements[1].clone(), List).values[1].clone(), Arithmetic);
+        assert_eq!(arith.op, ArithOp::Add);
+        let left_val = unwrap_fragment!(arith.left.unwrap().clone(), Arithmetic);
+        assert_eq!(left_val.op, ArithOp::Add);
+        let left_nested = unwrap_fragment!(left_val.left.unwrap().clone(), Raw);
+        assert_eq!(left_nested.value, "39");
+        let right_nested = unwrap_fragment!(left_val.right.unwrap().clone(), Raw);
+        assert_eq!(right_nested.value, "21");
+        let right = unwrap_fragment!(arith.right.unwrap().clone(), Raw);
+        assert_eq!(right.value, "80");
```

Comment:
You're not checking a type at any of these tests. I think you try to test it by checking if it's "80" and not "80.0" for example, but it's insufficient. What if we change in the future how we parse `Num` and we would strip floating point? Checking the actual type is the only way to be 100% sure it's correct